src/features/lightspeed/errors.ts

```diff
@@ -1,50 +1,86 @@
+import { AxiosError } from "axios";
 import { IError } from "../../interfaces/lightspeed";
 
 class Error implements IError {
   readonly code: string;
   readonly message?: string;
   readonly detail?: unknown;
+  readonly check: (err: AxiosError) => boolean;
 
-  public constructor(code: string, message?: string, detail?: unknown) {
+  private getCode(err: AxiosError): string {
+    const responseErrorData = <AxiosError<{ code?: string; message?: string }>>(
+      err?.response?.data
+    );
+    const code: string = responseErrorData.hasOwnProperty("code")
+      ? (responseErrorData.code as string)
+      : "unknown";
+    return code;
+  }
+
+  public constructor(
+    code: string,
+    message?: string,
+    detail?: unknown,
+    check?: (err: AxiosError) => boolean
+  ) {
     this.code = code;
     this.message = message;
     this.detail = detail;
-  }
-
-  public withMessage(message?: string): Error {
-    return new Error(this.code, message, this.detail);
+    if (check) {
+      this.check = check;
+    } else {
+      this.check = (err: AxiosError) => {
+        return this.code === this.getCode(err);
+      };
+    }
   }
 
   public withDetail(detail?: unknown): Error {
-    return new Error(this.code, this.message, detail);
+    // Clone the Error to preserve the original definition
+    return new Error(this.code, this.message, detail, this.check);
   }
 }
 
 class Errors {
-  private errors: Map<number, Array<IError>> = new Map();
+  private errors: Map<number, Array<Error>> = new Map();
 
-  public addError(statusCode: number, error: IError) {
+  public addError(statusCode: number, error: Error) {
     if (!this.errors.has(statusCode)) {
       this.errors.set(statusCode, []);
     }
     this.errors.get(statusCode)?.push(error);
   }
 
-  public getError(statusCode: number, code: string): Error | undefined {
-    if (!this.errors.has(statusCode)) {
+  public getError(err: AxiosError): Error | undefined {
+    if (err && !("response" in err)) {
```

Comment:
Testing content match with my server down (ECONNREFUSED), I end up with this:

![image](https://github.com/ansible/vscode-ansible/assets/3698180/331af2bc-40ff-4983-a91c-53247b0efbb3)

You can recreate this by working against your local service, accepting a completion request, taking your service down, then running "Fetch Training Matches" in the command palette.

Reply:
Oee. Did you pull the latest changes first?

You reported this `undefined` on my previous PR and I thought it was fixed in such a way that seeing `undefined` would be impossible.

Reply:
Doh. I just checked the code on `main` and it's eminently possible to still show `undefined`.

Reply:
Done.